Objects/tupleobject.c

```diff
@@ -860,6 +893,24 @@ static PyMappingMethods tuple_as_mapping = {
 
 static PyObject *tuple_iter(PyObject *seq);
 
+static void
+tuple_free(void *op)
+{
+    if (_PyTuple_HAS_NO_GC(op)) {
+        PyObject_Del(op);
+    }
+    else {
+        PyObject_GC_Del(op);
+    }
+}
+
+static int
+tuple_is_gc(PyObject *op)
```

Comment:
FWIW, I considered using `tp_is_gc` like this but was worried it would have too big an impact on performance.  Apparently that wasn't the case. 🙂

Reply:
Most tuples like (1,2) or ("foo","bar) are untracked at first GC cycle.
So calling one callback on each tuple is not a big problem.

When adding callback for list, dict, or function objects, we need to measure the performance impact.